src/main/java/org/spongepowered/api/item/inventory/ItemStack.java

```diff
@@ -135,6 +144,14 @@ static ItemStack of(ItemType itemType, int quantity) {
      */
     boolean equalTo(ItemStack that);
 
+    /**
+     * Returns true if {@link #getQuantity()} is zero and therefore this
+     * ItemStack is empty.
```

Comment:
Wouldn't that be implementation detail?
In vanilla it won't render it because empty ItemStacks are always of type AIR.
You CAN change the quantity back up to a positive number.
Then the Item is no longer of type AIR. (if it was not before).
Offering an empty Itemstack will do nothing.
But setting it is a valid thing to do.